### PR TITLE
fix(context): context nested grandchild

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. kick the context providers on each update, to ensure that even poorly hydrated elements get context

### CODE REVIEW

**Review by zeroedin — COMMENTED** [review]
image

Dark palette cards still contain a light theme cta on the pattern elements here: https://deploy-preview-1985--red-hat-design-system.netlify.app/patterns/card/

File: eleventy.config.ts [hunk 1]
```diff
@@ -57,15 +54,9 @@ export default async function(eleventyConfig: UserConfig) {
...
-    const { stdout, stderr } = await exec('npx tspc -b');
-    if (stderr) {
-      // eslint-disable-next-line no-console
-      console.error(stderr);
-    } else {
-      // eslint-disable-next-line no-console
-      console.log(stdout);
-    }
+  let watch;
+  eleventyConfig.on('eleventy.before', function() {
```
Comment (bennypowers) [inline eleventy.config.ts:58]:
```suggestion
  eleventyConfig.on('eleventy.before', function({ runMode }) {
```

File: eleventy.config.ts [hunk 2]
```diff
@@ -57,15 +54,9 @@ export default async function(eleventyConfig: UserConfig) {
...
+    watch ||= $({ stdout: ['pipe'], stderr: ['pipe'] })`npx tspc -b --watch`;
```
Comment (bennypowers) [inline eleventy.config.ts:59]:
```suggestion
    watch ||= run-mode === 'watch' && $({ stdout: ['pipe'], stderr: ['pipe'] })`npx tspc -b --watch`;
```

File: eleventy.config.ts [hunk 3]
```diff
@@ -57,15 +54,9 @@ export default async function(eleventyConfig: UserConfig) {
...
-    if (stderr) {
-      // eslint-disable-next-line no-console
-      console.error(stderr);
-    } else {
-      // eslint-disable-next-line no-console
-      console.log(stdout);
-    }
+  let watch;
+  eleventyConfig.on('eleventy.before', function({ runMode }) {
+    watch ||= run-mode === 'watch' && $({ stdout: ['pipe'], stderr: ['pipe'] })`npx tspc -b --watch`;
```
Comment (bennypowers) [inline eleventy.config.ts:59]:
```suggestion
    watch ||= runMode === 'watch' && $({ stdout: ['pipe'], stderr: ['pipe'] })`npx tspc -b --watch`;
```

File: eleventy.config.ts [hunk 4]
```diff
@@ -54,9 +54,10 @@ export default async function(eleventyConfig: UserConfig) {
     eleventyConfig.addGlobalData('runMode', runMode);
   });
 
-  let watch;
   eleventyConfig.on('eleventy.before', function({ runMode }) {
-    watch ||= runMode === 'watch' && $({ stdout: ['pipe'], stderr: ['pipe'] })`npx tspc -b --watch`;
+    (() => {
+      return runMode === 'watch' && $({ stdout: ['pipe'], stderr: ['pipe'] })`npx tspc -b --watch`;
```
Comment (bennypowers) [inline eleventy.config.ts:59]: This will spawn a new tsc watch process on each build

Comment (zeroedin) [reply eleventy.config.ts]: good thinking, was thinking the opposite at the time, reverted

**Review by zeroedin — APPROVED** [review]
Lurking gremlins tease mischief

**Pull request merged**